FirebaseAuth/Sources/Swift/Backend/AuthBackend.swift

```diff
@@ -185,24 +185,22 @@ private class AuthBackendRPCImplementation: NSObject, AuthBackendImplementation
 
   #if os(iOS)
     private class func generateMFAError(response: AuthRPCResponse, auth: Auth) -> Error? {
-      if let mfaResponse = response as? EmailLinkSignInResponse,
-         mfaResponse.idToken == nil,
-         let enrollments = mfaResponse.mfaInfo {
+      if let mfaResponse = response as? AuthMFAResponse,
+         mfaResponse.nilIDToken(),
```

Comment:
I don't fully understand the implications, but if it makes sense to you, should `AuthMFAResponse` be reworked so it never has a nil ID token? I was trying to think of a way to allow the compiler to enforce that an IDToken is never nil for select `*Response` types.

Reply:
The implications aren't clear to me either - and it would be a more fundamental change to start throwing from the Responses. 

It probably should be that way, but I think it would be a bigger rework than I want to bite off in the Swift reimplemenation.